flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Expressions.java

```diff
@@ -535,6 +535,18 @@ public static ApiExpression ifThenElse(Object condition, Object ifTrue, Object i
         return apiCall(BuiltInFunctionDefinitions.IF, condition, ifTrue, ifFalse);
     }
 
+    /**
+     * Coalesce specifies a series of expressions, and returns the first expression whose value is
+     * not null. If all the expressions evaluate as null, coalesce returns a null value.
+     *
+     * <p>e.g. coalesce($("f0"), "-") leads to "-" if f0 is null
```

Comment:
I think we can go into detail a bit more here and explain the return type produced by this function (common type, nullability). 

We can also format the example part better and more consistent (see some of the other recent examples like in `BaseExpressions`) by making use of a proper code block. Having a couple examples would be good, especially ones that don't rely on external fields (like f0) but would work out of the box.

Reply:
Also we should give the same documentation treatment to the function's description in the actual docs (https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/dev/table/functions/systemfunctions/).

Reply:
> Having a couple examples would be good, especially ones that don't rely on external fields (like f0) but would work out of the box.

I think it's still worth showing with external fields, because showing the coalesce with constants doesn't really show the point IMO. Look at the rephrase now, i have added two examples to make my point (hopefully) clear enough.